wire/cli/src/main.rs

```diff
@@ -49,7 +60,72 @@ async fn main() -> Result<(), anyhow::Error> {
             no_keys,
             always_build_local,
         } => {
+            let stdin = std::io::stdin();
+
+            // set fd to nonblocking so it won't hang the program in case if
+            // there is no piping to stdin
+            let fdflags = fcntl(stdin.as_raw_fd(), F_GETFL)?;
+            let newflags = OFlag::from_bits_retain(fdflags | O_NONBLOCK);
+            fcntl(stdin.as_raw_fd(), F_SETFL(newflags))?;
+
+            let mut handle = stdin.lock();
+
+            // This could be improved probably, but for now this stays.
+            let nodes = {
+                let mut buf: Vec<u8> = Vec::new();
+                let mut c: u8 = 0;
+                loop {
+                    match handle.read_until(b'\n', &mut buf) {
+                        Result::Ok(0) => break,
+                        Result::Ok(_) => {
+                            break;
+                        }
+                        Err(e) => match e.raw_os_error() {
+                            Some(11) => {
+                                // Resource temporarily unavailable is a normal
+                                // error response when reading from stdin
+                                // nonblocking
+                                //
+                                // as for if there should be a counter for this,
+                                // this is undecided.
+                                if c >= 2 {
+                                    break;
+                                }
+                                c += 1;
+                            }
+                            Some(code) => {
+                                bail!("unhandled stdin error: {code} {e}");
+                            }
+                            None => bail!(e),
+                        },
+                    }
+                }
+
+                String::from_utf8(buf)
+                    .unwrap()
+                    .trim_end()
+                    .split(',')
+                    .filter(|x| !x.is_empty())
+                    .map(|node| ApplyTarget::from(node.to_string()))
+                    .collect::<Vec<ApplyTarget>>()
+            };
+
+            let on = if let Some(n) = on {
+                if !nodes.is_empty() {
+                    bail!("ambiguous targets specified by parameter and stdin");
+                }
+                n
+            } else {
+                if nodes.is_empty() {
+                    bail!("no provided targets found");
+                }
+                nodes
+            };
```

Comment:
use unwrap_or

Reply:
Where?

Reply:
Can this be written as this?

```suggestion

            if !nodes.is_empty() && on.is_some() {
                warn!("--on is ignored when stdin is passed");
            }
            
            let on = on.unwrap_or(node);
```

Reply:
oh yeah I guess so, thanks

Reply:
well wait, do you really want to have precedence here? both cli provided and stdin provided hold the same level of precedence in my opinion, so there is ambiguity with what should be accepted.